calico/felix/ipsets.py

```diff
@@ -127,11 +122,10 @@ def cleanup(self):
         all_ipsets = list_ipset_names()
         # only clean up our own rubbish.
         pfx = IPSET_PREFIX[self.ip_type]
-        tmppfx = IPSET_TMP_PREFIX[self.ip_type]
-        felix_ipsets = set([n for n in all_ipsets if n.startswith(pfx) or
-                                                     n.startswith(tmppfx)])
+        felix_ipsets = set([n for n in all_ipsets if n.startswith(pfx)])
         whitelist = set()
-        for ipset in self.objects_by_id.values():
+        for ipset in (self.objects_by_id.values() +
+                      self.stopping_objects_by_id.values()):
```

Comment:
If there's a stopping ActiveIpset, it will be trying to do its own cleanup.  I thought this might have been the bug that Nick was hitting but that wasn't borne out by his logs.